### ISSUE TITLE
Incorrect CA1854 + CSharpPreferDictionaryTryGetValueFixer crashing

### ISSUE THREAD

**Issue by Yozer** [dotnet/roslyn-analyzers#6273]
### Analyzer

**Diagnostic ID**: [CA1854](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1854): `Prefer the IDictionary.TryGetValue(TKey, out TValue) method`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet7/NuGet/Microsoft.CodeAnalysis.NetAnalyzers/overview/7.0.0-preview1.22559.1

### Describe the bug

1. Incorrect CA1854
2. CSharpPreferDictionaryTryGetValueFixer is crashing:
```
System.InvalidOperationException : GetCurrentNode returned null with the following node: cache[1] - line 288
   at Roslyn.Utilities.Contract.Fail(String message,Int32 lineNumber)
   at Microsoft.CodeAnalysis.Editing.SyntaxEditor.GetChangedRoot()
   at async Microsoft.NetCore.CSharp.Analyzers.Runtime.CSharpPreferDictionaryTryGetValueFixer.<>c__DisplayClass1_0.<RegisterCodeFixesAsync>b__0(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
```

### Steps To Reproduce

```csharp
var cache = new Dictionary<int, Dictionary<string, string>>();

if (cache[1].ContainsKey("test"))
{
    Console.WriteLine("Hello, World!");
}
```

### Expected behavior
CA1854 should not be reported at all.
### Actual behavior
CA1854  being reporter + unable to apply fix because of the crash.
## Additional context

Visual Studio: 16.4
.NET 7
```<AnalysisLevel>latest-Minimum</AnalysisLevel>```
